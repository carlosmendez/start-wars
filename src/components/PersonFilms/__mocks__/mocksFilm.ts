export const mockFilms = [
  {
    "title": "mockFilm1",
    "releaseDate": "mockReleaseDate1",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 40
        },
        {
          "surfaceWater": 8
        }
      ]
    },
    "producers": [
      "Gary Kurtz",
      "Rick McCallum"
    ]
  },
  {
    "title": "mockFilm2",
    "releaseDate": "mockReleaseDate2",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 100
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": 10
        }
      ]
    },
    "producers": [
      "Gary Kurtz",
      "Rick McCallum"
    ]
  },
  {
    "title": "mockFilm3",
    "releaseDate": "mockReleaseDate3",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 12
        },
        {
          "surfaceWater": null
        }
      ]
    },
    "producers": [
      "Howard G. Kazanjian",
      "George Lucas",
      "Rick McCallum"
    ]
  },
  {
    "title": "4",
    "releaseDate": "mockReleaseDate4",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 40
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 12
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": 0.9
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": 60
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        }
      ]
    },
    "producers": [
      "Rick McCallum"
    ]
  }
];
