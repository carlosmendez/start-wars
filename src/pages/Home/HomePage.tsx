import React from 'react';
import { gql, useQuery } from "urql";
import { MainTitle } from "../../components/MainTitle";
import { StyledGrid, StyledGridItem, StyledLink } from "./styled-components";

const query = gql`
  query Home {
    allPeople {
      edges {
        node {
          id
          name
        }
      }
    }
  }
`;

const HomePage = () => {
  const [result] = useQuery({ query });
  const { data, fetching, error } = result;

  if (fetching) return <p>Loading...</p>;
  if (error) return <p>Oh no... {error.message}</p>;

  return (
    <>
      <MainTitle>People</MainTitle>
      <StyledGrid>
        {/*TODO: add types to 'item'*/}
        {data && data.allPeople.edges.map((item: any, index: number) => (
          <StyledGridItem key={index}>
            <StyledLink to={`/person/${item.node.id}`}>
              {item.node.name}
            </StyledLink>
          </StyledGridItem>
        ))}
      </StyledGrid>
    </>
  );
};

export default HomePage;
