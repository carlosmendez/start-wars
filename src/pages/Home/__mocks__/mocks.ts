export const mockPeople = {
  "allPeople": {
    "edges": [
      {
        "node": {
          "name": "Luke Skywalkexxr",
          "id": "cGVvcGxlOjE=",
          "species": 50
        }
      },
      {
        "node": {
          "name": "C-3PO",
          "id": "cGVvcGxlOjI=",
          "species": {
            "averageHeight": null
          }
        }
      }
    ]
  }
};
