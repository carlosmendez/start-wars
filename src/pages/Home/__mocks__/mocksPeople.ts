export const mockPeople = {
  "allPeople": {
    "edges": [
      {
        "node": {
          "name": "mockName1",
          "id": "mockId1",
          "species": 50
        }
      },
      {
        "node": {
          "name": "mockName2",
          "id": "mockId2",
          "species": {
            "averageHeight": null
          }
        }
      }
    ]
  }
};
