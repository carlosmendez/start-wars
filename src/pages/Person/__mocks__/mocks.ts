export const mockPerson = {
  "person": {
    "name": "Luke Skywalker",
      "birthYear": "19BBY",
      "species": null,
      "filmConnection": {
      "films": [
        {
          "title": "A New Hope",
          "releaseDate": "1977-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              }
            ]
          },
          "producers": [
            "Gary Kurtz",
            "Rick McCallum"
          ]
        },
        {
          "title": "The Empire Strikes Back",
          "releaseDate": "1980-05-17",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 100
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 10
              }
            ]
          },
          "producers": [
            "Gary Kurtz",
            "Rick McCallum"
          ]
        },
        {
          "title": "Return of the Jedi",
          "releaseDate": "1983-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 12
              },
              {
                "surfaceWater": null
              }
            ]
          },
          "producers": [
            "Howard G. Kazanjian",
            "George Lucas",
            "Rick McCallum"
          ]
        },
        {
          "title": "Revenge of the Sith",
          "releaseDate": "2005-05-19",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 12
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": 0.9
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 60
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              }
            ]
          },
          "producers": [
            "Rick McCallum"
          ]
        }
      ],
        "pageInfo": {
        "hasNextPage": false,
          "hasPreviousPage": false,
          "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
          "endCursor": "YXJyYXljb25uZWN0aW9uOjM="
      },
      "totalCount": 4
    }
  }
};
