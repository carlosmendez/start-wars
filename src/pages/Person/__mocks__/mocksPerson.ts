export const mockPerson = {
  "person": {
    "name": "mockName1",
      "birthYear": "mockBirthYear",
      "species": null,
      "filmConnection": {
      "films": [
        {
          "title": "mockFilm1",
          "releaseDate": "1977-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              }
            ]
          },
          "producers": [
            "mockProducer1",
            "mockProducer2"
          ]
        },
        {
          "title": "mockFilm2",
          "releaseDate": "1980-05-17",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 100
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 10
              }
            ]
          },
          "producers": [
            "mockProducer1",
            "mockProducer3"
          ]
        },
      ],
        "pageInfo": {
        "hasNextPage": false,
          "hasPreviousPage": false,
          "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
          "endCursor": "YXJyYXljb25uZWN0aW9uOjM="
      },
      "totalCount": 2
    }
  }
};
