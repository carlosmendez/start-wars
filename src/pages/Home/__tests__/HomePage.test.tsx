import React from 'react';
import renderer from 'react-test-renderer';
import { Provider, CombinedError } from 'urql';
import { never, fromValue } from 'wonka';
import HomePage from "../HomePage";
import { BrowserRouter as Router } from 'react-router-dom';
import { mockPeople} from "../__mocks__/mocks";

describe('Home Page component', () => {
  it('It renders correctly with response Success', () => {
    const successState = {
      executeQuery: () =>
        fromValue({
          data: {
            ...mockPeople
          }
        }),
    };

    const jsx = (
      <Provider value={successState}>
        <Router>
          <HomePage />
        </Router>
      </Provider>
    );
    const tree = renderer
      .create(jsx)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it is fetching with Response error', () => {
    const errorState = {
      executeQuery: () =>
        fromValue({
          error: new CombinedError({
            networkError: Error('something went wrong!'),
          }),
        }),
    };

    const jsx = (
      <Provider value={errorState}>
        <HomePage />
      </Provider>
    );
    const tree = renderer
      .create(jsx)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it is calling fetch', () => {
    const fetchingState = {
      executeQuery: jest.fn(() => never),
    };
    const jsx = (
      <Provider value={fetchingState}>
        <HomePage />
      </Provider>
    );
    renderer
      .create(jsx)
      .toJSON();

    expect(fetchingState.executeQuery).toBeCalledTimes(1);
  });

});
