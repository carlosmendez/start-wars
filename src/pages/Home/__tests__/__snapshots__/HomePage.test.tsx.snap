// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page component It renders correctly with response Success 1`] = `
[
  <h1
    className="css-1yltisf"
  >
    Star Wars characters
  </h1>,
  <div
    className="css-7ydzf1"
  >
    <div
      className="css-16d4tpu"
    >
      <a
        className="css-iwdcak"
        href="/person/cGVvcGxlOjE="
        onClick={[Function]}
      >
        Luke Skywalkexxr
      </a>
    </div>
    <div
      className="css-16d4tpu"
    >
      <a
        className="css-iwdcak"
        href="/person/cGVvcGxlOjI="
        onClick={[Function]}
      >
        C-3PO
      </a>
    </div>
  </div>,
]
`;

exports[`Home Page component it is fetching with Response error 1`] = `
<p>
  [Network] something went wrong!
</p>
`;
